src/app/services/project_service.js

```diff
@@ -221,10 +223,32 @@ angular
                 return _.includes(['model', 'source', 'seed', 'snapshot', 'analysis', 'exposure'], node.resource_type);
             });
 
-            service.project.searchable = _.filter(search_nodes.concat(search_macros), function(obj) {
+            service.project.searchable = prepareModelsForSearching(_.filter(search_nodes.concat(search_macros), function(obj) {
                 // It should not be possible to search for hidden documentation
                 return !obj.docs || obj.docs.show;
-            });
+            }));
+
+			const fuseOptions = {
+			  includeScore: true,
+			  includeMatches: true,
+			  ignoreLocation: true,
+			  //useExtendedSearch: true,
```

Comment:
I really want to enable extended search, because it means you can prefix things with a single quote to force exact matches over fuzzy ones. However when I enabled it, I found that the matches array has _way_ more sharp edges (to name a few: unsorted, duplicate records, overlapping ranges). 

The duplicates and sorting are pretty trivial to resolve, but I haven't come up with an elegant solution to making sense of [[1, 3], [1, 1], [1, 2], [2, 3]]. I think this might be something to come back and have a second pass at once the bulk of it's merged in.

Reply:
right on - totally agree - let's get fuse in here and then make it more sophisticated in the future!